simulator/src/main.rs

```diff
@@ -415,3 +464,19 @@ async fn run_configured_simulation(cli: &Cli, producer: &FutureProducer, defined
         }
     }
 }
+
+fn init_tracer(otel_endpoint: Option<&str>) -> Option<OtelTracer> {
```

Comment:
This can be made sufficiently generic to go into `common` with only adding a couple of parameters.

In fact, if you made it a macro you could get away with the application name being taken from the crate name.